src/WebJobs.Script/Description/Workers/WorkerFunctionDescriptorProvider.cs

```diff
@@ -65,11 +65,15 @@ protected override IFunctionInvoker CreateFunctionInvoker(string scriptFilePath,
         {
             var bindings = inputBindings.Union(outputBindings);
 
-            // If an input or output binding uses expressions, we do not want to use ParameterBindingData for the trigger bindings
             if (triggerMetadata.SupportsDeferredBinding())
             {
-                bool skipDeferredBinding = BindingAttributeContainsExpression(bindings);
-                if (skipDeferredBinding)
+                // If an input or output binding uses expressions, we do not want to use ParameterBindingData for the trigger bindings
+                bool containsExpression = BindingAttributeContainsExpression(bindings);
+
+                // If the trigger is a CosmosDBTrigger, we do not want to use ParameterBindingData as it is not supported
+                bool isCosmosTrigger = IsCosmosDBTrigger(triggerMetadata);
```

Comment:
Yup agree, was trying to have a quick patch in place for if we release Cosmos soon.

However, I did sync with Fabio about this today and since Cosmos is not in any rush to be released we can focus on refactoring the supportsDeferredBinding attribute for extensions and the function metadata generator on the worker to have more granular information about what bindings support deferred binding rather than the blanket "all or nothing" approach we have in place today. 

As a result, I'll abandon this PR and create an issue for the refactor work that needs to happen in the worker